### PR TITLE
More useful error message for when DOB is entered incorrectly.

### LINKED ISSUE #1010: Can we have a more descriptive error message when WCA ID claiming faces DOB mismatch?
It presently says "incorrect".  I propose "The provided date of birth doesn't match our database. If you're sure you're providing the correct date in the correct format (YYYY-MM-DD) please email the WCA Results Team (link) reporting this error and sending a picture of any legal document (identity card, driver license,...) where your name and date of birth are visible."

### PR DESCRIPTION
**Pull request by jfly**
This fixes #1010.

This should not be merged until after #1013.

### CODE REVIEW

File: WcaOnRails/config/locales/en.yml [hunk 1]
```diff
@@ -436,14 +436,14 @@ en:
     errors:
       not_found: "not found"
       already_assigned: "already assigned to a different user"
-      dob_incorrect: "incorrect"
+      dob_incorrect_html: "The provided date of birth doesn't match our database. If you're sure you're providing the correct date in the correct format (YYYY-MM-DD) please email the <a href='mailto:results@worldcubeassociation.org' target='_blank'>WCA Results Team</a> reporting this error and sending a picture of any legal document (identity card, driver license,...) where your name and date of birth are visible."
```
Comment (larspetrus) [inline WcaOnRails/config/locales/en.yml]: "...and send a picture..." is better English.

File: WcaOnRails/config/locales/en.yml [hunk 2]
```diff
@@ -436,14 +436,14 @@ en:
...
       unique: "must be unique"
       dob_past: "must be in the past"
       must_be_senior: "must be a senior delegate"
       must_not_be_present: "must not be present"
       senior_has_delegate: "cannot demote senior delegate with subordinate delegates"
       already_have_id: "cannot claim a WCA ID because you already have WCA ID %{wca_id}"
-      wca_id_no_birthdate: "WCA ID does not have a birthdate. Contact the Results team to resolve this."
+      wca_id_no_birthdate_html: "WCA ID does not have a birthdate. Please email the <a href='mailto:results@worldcubeassociation.org' target='_blank'>WCA Results Team</a> reporting this error and sending a picture of any legal document (identity card, driver license,...) where your name and date of birth are visible."
```
Comment (larspetrus) [inline WcaOnRails/config/locales/en.yml]: "... report this error and send..."

File: WcaOnRails/spec/features/sign_up_spec.rb [hunk 3]
```diff
@@ -52,7 +52,7 @@
 
       # Make sure we inform the user of the incorrect birthdate they just
       # entered.
-      expect(page.find(".alert.alert-danger")).to have_content("Birthdate incorrect")
+      expect(page.find(".alert.alert-danger")).to have_content("Birthdate The provided date of birth doesn't match our database.")
```
Comment (larspetrus) [inline WcaOnRails/spec/features/sign_up_spec.rb]: That's not a sentence.

File: WcaOnRails/spec/models/user_spec.rb [hunk 4]
```diff
@@ -345,7 +345,8 @@
       other_person = FactoryGirl.create :person, year: 1980, month: 2, day: 1
       user.unconfirmed_wca_id = other_person.wca_id
       expect(user).to be_invalid
-      expect(user.errors.messages[:dob_verification]).to eq ['incorrect']
+      expect(user.errors.messages[:dob_verification].length).to eq 1
+      expect(user.errors.messages[:dob_verification][0]).to start_with "The provided date of birth doesn't match our database."
```
Comment (larspetrus) [inline WcaOnRails/spec/models/user_spec.rb]: Hm... The much better way is to expect the translation key (`"dob_incorrect_html"`). As I remember it, it's neither trivial or very hard to do, but I don't remember how we did it :)

It's really worth having in an internationalized project, but is probably for a separate PR.

File: WcaOnRails/config/locales/en.yml (hunk 1 above)
Comment (jfly) [inline WcaOnRails/config/locales/en.yml]: Done

File: WcaOnRails/config/locales/en.yml (hunk 2 above)
Comment (jfly) [inline WcaOnRails/config/locales/en.yml]: Ahh, I think that change forces us to need a comma, but I'm pretty bad at this stuff. Please tell me what you think of the updated version.

File: WcaOnRails/spec/models/user_spec.rb (hunk 4 above)
Comment (jfly) [inline WcaOnRails/spec/models/user_spec.rb]: Hmm... one easy fix in this case is to expect `I18n.t('users.errors.dob_incorrect_html')`. I think that's better than what we have now. Let me know what you think!

**Comment by larspetrus** [comment]
Yeah, if expecting `I18n.t('users.errors.dob_incorrect_html')` works, that fulfills all my wishes here.

File: WcaOnRails/config/locales/en.yml [hunk 5]
```diff
@@ -436,14 +436,14 @@ en:
...
-      dob_incorrect: "incorrect"
+      dob_incorrect_html: "does not match our database. If you're sure you're providing the correct date in the correct format (YYYY-MM-DD) please email the <a href='mailto:results@worldcubeassociation.org' target='_blank'>WCA Results Team</a>, report this error and send a picture of any legal document (identity card, driver license,...) where your name and date of birth are visible."
       unique: "must be unique"
       dob_past: "must be in the past"
       must_be_senior: "must be a senior delegate"
       must_not_be_present: "must not be present"
       senior_has_delegate: "cannot demote senior delegate with subordinate delegates"
       already_have_id: "cannot claim a WCA ID because you already have WCA ID %{wca_id}"
-      wca_id_no_birthdate: "WCA ID does not have a birthdate. Contact the Results team to resolve this."
+      wca_id_no_birthdate_html: "WCA ID does not have a birthdate. Please email the <a href='mailto:results@worldcubeassociation.org' target='_blank'>WCA Results Team</a>, report this error and send a picture of any legal document (identity card, driver license,...) where your name and date of birth are visible."
```
Comment (jfly) [inline WcaOnRails/config/locales/en.yml]: How does this revision sound?

> We do not have a birthdate recorded for this WCA ID. Please email the <a href='mailto:results@worldcubeassociation.org' target='_blank'>WCA Results Team</a> with a photo of any legal document (ID card, driver license, etc) where your name and date of birth are visible.

I never knew about the difference between license and licence. As the WCA is going to incorporate in the US, we do want to go with American English for everything, so I think license is the right word here.

Comment (larspetrus) [inline WcaOnRails/config/locales/en.yml]: I can't find a thing to complain about in that!

Comment (jfly) [inline WcaOnRails/config/locales/en.yml]: Thanks for the feedback! I've updated the PR with the latest.

**Comment by jfly** [comment]
I plan to merge this up on Friday unless someone objects before then.

**Pull request merged**